test/SlowTests/Server/Documents/ETL/EtlCountersTests.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Server.Config;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Server.Documents.ETL
+{
+    public class EtlCountersTests : EtlTestBase
+    {
+        private const int _waitInterval = 1000;
+
+        private readonly Options _options = Debugger.IsAttached
+            ? new Options {ModifyDatabaseRecord = record => record.Settings[RavenConfiguration.GetKey(x => x.Etl.ExtractAndTransformTimeout)] = "300"}
+            : null;
+        
+        public EtlCountersTests(ITestOutputHelper output) : base(output)
+        {
+        }
+        
+        [Theory]
+        [InlineData(16)]
+        [InlineData(1024)]
+        public async Task EtlCounter_WhenUseAddCountersAndRemoveCounterFromSrc_ShouldRemoveTheCounterFromDest(int count)
+        {
+            using var src = GetDocumentStore(_options);
+            try
+            {
+                using var dest = GetDocumentStore();
+
+                var etlResult = AddEtl(src, dest, "Users", script: @"
+var doc = loadToUsers(this);
+var counters = this['@metadata']['@counters'];
+for (var i = 0; i < counters.length; i++) {
+    doc.addCounter(loadCounter(counters[i]));
+}");
+                var entity = new User();
+                var counters = Enumerable.Range(0, count).Select(i => "Likes" + i).ToArray();
+                
+                using (var session = src.OpenAsyncSession())
+                {
+                    await session.StoreAsync(entity);
+
+                    foreach (var counter in counters)
+                    {
+                        session.CountersFor(entity.Id).Increment(counter);
+                    }
+                    await session.SaveChangesAsync();
+                }
+                await AssertWaitForTrueAsync(async () =>
+                {
+                    using var session = dest.OpenAsyncSession();
+                    var all = (await session.CountersFor(entity.Id).GetAllAsync()).Keys.ToArray();
+                    return all.Except(counters).Any() == false;
+                }, interval:_waitInterval);
+
+                using (var session = src.OpenAsyncSession())
+                {
+                    foreach (var counter in counters)
+                    {
+                        session.CountersFor(entity.Id).Delete(counter);
+                    }
+                    await session.SaveChangesAsync();
+                }
+                await AssertWaitForTrueAsync(async () =>
+                {
+                    using var session = dest.OpenAsyncSession();
+                    return (await session.CountersFor(entity.Id).GetAllAsync()).Any() == false;
+                }, interval:_waitInterval);
+            }
+            catch (Exception e)
+            {
+                ThrowWithEtlErrors(src, e);
```

Comment:
Try catch on entire test method looks a bit weird. I understand you want to include some additional info but is it really needed? That's handy for debugging a problem with failing test but I think it's more important to immediately see which assertion has failed.

Reply:
I found it very helpful. 
If you got failing test without the Etl error you have to rerun the test.
At best, it will just time-consuming but if the test becomes failing in once a couple of runnings the extra info is very important.
In the end, the final error includes the exact assert but you right it is not immediately.

Reply:
yeah, I understand your point but this is very strange. Can we handle it somehow better? Maybe do something in the base class `EtlTestBase` ?